### PR TITLE
feat: allow to set a URI

### PR DESCRIPTION
**Pull request by Alex-Werner**
## Issue being fixed or feature implemented

In usage of V5, we used to use ZMQ by passing a path, this allow to use either URI or basic host/port pair. 

## What was done?

- added ability to pass a uri as constructor params.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: lib/ZMQClient/methods/connect.js [hunk 1]
```diff
@@ -1,56 +1,81 @@
...
-    await this.disconnect();
+    await this.disconnect(uri);
   }
-  const uri = `${this.protocol}://${this.host}:${this.port}`;
 
   const opts = { ...socketOption };
 
+  if (opts.maxRetries) {
+    // Subscriber opts is not extensible, we need to remove that from opts we pass
+    delete opts.maxRetries;
```
Comment (antouhou) [inline lib/ZMQClient/methods/connect.js:28]: I'm not sure how deleting it helps?

Comment (Alex-Werner) [reply lib/ZMQClient/methods/connect.js:28]: Passing `opts` with a property named `maxRetries` to new Subscriber() would result in Subscriber to error out due to not being a valid props.

**Comment by Alex-Werner** [comment]
Rebased onto master.

File: lib/ZMQClient/methods/connect.js (hunk 1 above)
Comment (Alex-Werner) [reply lib/ZMQClient/methods/connect.js]: Fixed !

**Review by antouhou — APPROVED** [review]

**Pull request merged**